core/llm_docs.md

```diff
@@ -0,0 +1,227 @@
+The Dagger tool calling system exposes its GraphQL API through a dynamic set of tools. The available tools and objects may change throughout your interaction based on context and previous operations.
+
+## Key Mechanics
+
+The `selectTools` tool describes available tools and objects, allowing you to select more tools at any time.
+
+All tools interact with Objects referenced by IDs in the form `ObjectType#123` (e.g., Container#1, File#2). Parameters named after object types refer to specific object instances of that type.
+
+All objects are immutable. Each tool call creates a new object as output that becomes available for subsequent operations.
+
+A tool called `save` may be present. If available, you MUST call it after completing the requested task.
+
+## Conceptual Framework
+
+Think of this system as a chain of transformations where each operation:
+1. Takes one or more immutable objects as input
+2. Performs a transformation according to specified parameters
+3. Returns a new immutable object as output
+4. Makes this new object available for subsequent operations
+
+## Example
+
+Below is an example usage of the tool calling scheme. Note that all potato and
+cooking-related concepts are placeholders. In practice there will be entirely
+different types of objects and APIs.
+
+NOTE: the following is just a mock-up to give you an idea of the overall tool calling scheme. Don't pay any attention to the particular syntax.
+
+<example>
+  <user>
+    <tools>
+      # Gain new tools for subsequent turns.
+      #
+      # Available objects:
+      # - Oven#1: An oven you can use for cooking.
+      # - Potato#1: The raw potato to turn into french fries.
+      # - Sink#1: A sink for cleaning food.
+      # - Plate#1: A plate for serving food.
+      #
+      # Available tools:
+      # - Oven_bake
+      # - Oven_broil
+      # - Potato_dice
+      # - Potato_peel
+      # - Potato_slice
+      # - Sink_rinse
+      # - Fries_plate
```

Comment:
is this commented section representative of a dynamically returned tool description? like on each turn we change how the selectTools tool describes itself?

Reply:
Yep exactly.